src/components/NewVacancies/index.jsx

```diff
@@ -0,0 +1,187 @@
+        applications: 4,
+        company: 'Company 3',
+        date: new Date(2022, 1, 1),
+    },
+];
+
+export const NewVacancies = () => {
+    const [page, setPage] = React.useState(0);
+    const [rowsPerPage, setRowsPerPage] = React.useState(5);
+
+    // Avoid a layout jump when reaching the last page with empty rows.
+    const emptyRows = page > 0 ? Math.max(0, (1 + page) * rowsPerPage - vacancies.length) : 0;
+
+    const handleChangePage = (event, newPage) => {
+        setPage(newPage);
+    };
+
+    const handleChangeRowsPerPage = (event) => {
+        setRowsPerPage(parseInt(event.target.value, 10));
+        setPage(0);
+    };
+
+    return (
+        <TableContainer component={Paper} elevation={3}>
+            <Typography variant="h2" sx={{ p: 2 }}>
+                Last vacancies created
+            </Typography>
+            <Table sx={{ minWidth: 500 }} aria-label="custom pagination table">
+                <TableBody>
+                    {(rowsPerPage > 0
+                        ? vacancies.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+                        : vacancies
+                    ).map((vacancy) => (
+                        <TableRow key={vacancy.id}>
+                            <TableCell component="th" scope="row">
+                                <VacancyItemCard
+                                    vacancy={vacancy.vacancy}
+                                    applications={vacancy.applications}
+                                    date={moment(vacancy.date).format('DD/MM/YYYY')}
+                                />
+                            </TableCell>
+                        </TableRow>
+                    ))}
+
+                    {emptyRows > 0 && (
+                        <TableRow style={{ height: 53 * emptyRows }}>
+                            <TableCell colSpan={6} />
+                        </TableRow>
+                    )}
+                </TableBody>
+                <TableFooter>
+                    <TableRow>
+                        <TablePagination
+                            rowsPerPageOptions={[{ label: 'All', value: -1 }]}
+                            colSpan={3}
+                            count={vacancies.length}
+                            rowsPerPage={rowsPerPage}
+                            page={page}
+                            SelectProps={{
+                                inputProps: {
+                                    'aria-label': 'rows per page',
+                                },
+                                native: true,
+                            }}
+                            onPageChange={handleChangePage}
+                            onRowsPerPageChange={handleChangeRowsPerPage}
+                            ActionsComponent={TablePaginationActions}
+                        />
+                    </TableRow>
+                </TableFooter>
+            </Table>
+        </TableContainer>
+    );
+};

```

Comment:
You have two functions in the same component. I think you can split them

Reply:
The material UI doc handles these two functions in the same component